source/StartConditions.cpp

```diff
@@ -187,66 +181,19 @@ void StartConditions::Load(const DataNode &node)
 
 
 
-void StartConditions::LoadState(const DataNode &node, StartState state, bool coreData)
-{
-	bool clearDescription = !infoByState[state].description.empty();
-	for(const auto &child : node)
-	{
-		// If this is loading into unlocked state, account for ccoreStartData.
-		if(coreData)
-			if(CoreStartData::LoadChild(child, false, false))
-			{
-				if(child.Token(0) == "system" && child.Size() >= 2)
-					infoByState[StartState::UNLOCKED].system = child.Token(1);
-				else if(child.Token(0) == "planet" && child.Size() >= 2)
-					infoByState[StartState::UNLOCKED].planet = child.Token(1);
-				continue;
-			}
-
-		if(child.Token(0) == "name" && child.Size() >= 2)
-			infoByState[state].name = child.Token(1);
-		else if(child.Token(0) == "description" && child.Size() >= 2)
-		{
-			if(clearDescription)
-			{
-				infoByState[state].description.clear();
-				clearDescription = false;
-			}
-			infoByState[state].description += child.Token(1) + "\n";
-		}
-		else if(child.Token(0) == "thumbnail" && child.Size() >= 2)
-			infoByState[state].thumbnail = SpriteSet::Get(child.Token(1));
-		else if(child.Token(0) == "system" && child.Size() >= 2)
-			infoByState[state].system = child.Token(1);
-		else if(child.Token(0) == "planet" && child.Size() >= 2)
-			infoByState[state].planet = child.Token(1);
-	}
-}
-
-
-
-void StartConditions::FillState(StartState from, StartState to)
-{
-	if(!infoByState[to].thumbnail)
-		infoByState[to].thumbnail = infoByState[from].thumbnail;
-	if(infoByState[to].name.empty())
-		infoByState[to].name = infoByState[from].name;
-	if(infoByState[to].description.empty())
-		infoByState[to].description = infoByState[from].description;
-	if(infoByState[to].system.empty())
-		infoByState[to].system = infoByState[from].system;
-	if(infoByState[to].planet.empty())
-		infoByState[to].planet = infoByState[from].planet;
-}
-
-
-
 // Finish loading the ship definitions.
 void StartConditions::FinishLoading()
 {
 	for(Ship &ship : ships)
 		ship.FinishLoading(true);
 
+	// The UNLOCKED StartInfo should always display the correct information. Therefore, we get the
+	// planet and system names now. If we had gotten these during Load, the planet and system provided
+	// may now be invalid, meaning the CoreStartData would actually send the start to New Boston instead
+	// of what was displayed.
+	infoByState[StartState::UNLOCKED].planet = GetPlanet().Name();
+	infoByState[StartState::UNLOCKED].system = GetSystem().Name();
+
```

Comment:
so other states could just have empty info, right? Shouldnt we make sure some parts always are filled with smth; like the name of the start? (even if it is ???)

Reply:
I've moved the FillState calls down below here, because this is when we can guarantee that the unlocked info has everything it needs in order to fill the other states,

Reply:
seems good then

Reply:
it is like this:
unlocked fills revealed
revealed fills display

Thats what derpy wanted

Reply:
feels weird to me honestly I'd have expected smth going the other way around
low level filling high level not the opposite but maybe I'm reading this wrong and thats what is happening

the lowest level is revealed isnt it? it shouldnt be filled by unlocked

what I'd expect is this:

lowest level defaults to ??? and no fields
it fills up the highest levels one by one

Reply:
unlocked is the lowest level, with unlocked you can use the start

Reply:
well reverse my whole logic then since we dont have the same conventions for lowest/highest
but still you get what I mean?

Reply:
Yea, I get what I mean, but I need @Amazinite opinion on this, as he requested it like this.

Reply:
The important part is that no state should have empty data (aside from the thumbnail, which we can't give a reasonable default for aside from nullptr). Beyond that, I don't think that we can say one or another way to fill in the missing data is better or not, as you're making assumptions about how a content creator might want to display information about a start depending on its state. Sure, you might want to fill all the info with "???," but another creator might instead want to use cryptic hints to locations. Since we can't decide what exactly someone might want, I've opted to simply use whatever is already given to us to fill in any missing information in other states.

The reason that the filling goes unlocked -> revealed -> displayed is because unlocked was already the only state that we filled in missing information for; if no name, use "(Unnamed start)," if no description, use ("No description provided.)," if no planet, use "New Boston," and if no system, use "Rutilicus."

Reply:
I see what you mean with it was the previous default but unless ppl change smth it will always be unlocked and so they wont interract with this at all.
Ofc we're making assumptions but we shouldnt assume filling stuff for the CC that comes from higher levels; its not instinctive that by default unlocked info will display in revealed info to me.
We could always change this later on though I guess.

Reply:
Yea, I think a cc's perspective on this would be nice.

Reply:
My perspective is that it depends on the start; or more largely, on our attitude towards starts.

Are starts, overall, something to be kept secret until the appropriate time? Or are starts something to tease the player with whose details get revealed over time?

Likewise, are we prioritizing preventing spoilers. or having alternate starts?

If the former, then the default should be the "???" in every field, and the default image should be the question mark. Fields get overriden by stuff from higher states as they are unlocked. If something is missing, broken, or whatever, then it's going to be the default "???" This is because the default assumption will be "If there's nothing to display here, it's because we haven't gotten to the trigger to reveal something to put here yet."

On the flip side, if the priority is on having alternate starts, and just happens that some of them are hidden; then we should operate on the basis that the default values are to reveal everything for the start, and stuff is hidden by "???" and the question mark icon as appropriate for the reveal state. This means that, if there's a mistake somewhere, the start is going to default to being visible. This approach would probably have blanks as the default content, since the assumption is "If there's nothing to display here, it is because this start has nothing to put here."

---

I don't have a strong opinion either way, but I lean towards the secretive approach, where everything defaults to ??? until explicitly overriden by one of the reveals to make stuff visible. In other words, we start from the bottom, displaying absolutely nothing, and gradually add content to the display until the player can see everything.

Reply:
Hmmm, either way it's not really important as people can just define the info to ??? If they want to.

Reply:
True, but the root of the question is "what is the default / fallback state?" Does it "fallback" to the completely revealed state? Or does it fallback to something that looks like the completely concealed state? The latter makes more sense as a default state, since if it was working properly it'd be showing whatever the content creator wanted it to be; so the default-is-??? makes more sense from an overall standpoint.

Reply:
Just changed the REVEALED and DISPLAYED to fill in with "???", although they still take the thumbnail from the unlocked state since we can't provide a default otherwise.

Reply:
We can just use the outfit/unknown thumbnail as default

Reply:
We can't guarantee that image exists.